lib/ts/recipe/emailverification/recipeImplementation.ts

```diff
@@ -0,0 +1,184 @@
+                    });
+                }
+            );
+
+            if (json.status === "GENERAL_ERROR") {
+                throw new Error(json.message);
+            }
+
+            if (json.status !== "OK" && json.status !== "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR") {
+                return {
+                    status: "CUSTOM_RESPONSE",
+                    fetchResponse,
+                };
+            }
+
+            return {
+                status: json.status,
+                fetchResponse,
+            };
+        },
+
+        isEmailVerified: async function ({
+            config,
+            options,
+        }: {
+            config: NormalisedInputType;
+            options?: RecipeFunctionOptions;
+        }): Promise<
+            | {
+                  status: "OK";
+                  isVerified: boolean;
+                  fetchResponse: Response;
+              }
+            | {
+                  status: "CUSTOM_RESPONSE";
+                  fetchResponse: Response;
+              }
+        > {
+            const { json, fetchResponse } = await querier.get<{ status: "OK"; isVerified: boolean } | APIGeneralError>(
+                EMAIL_VERIFY_PATH,
+                {},
+                undefined,
+                async (context) => {
+                    let postRecipeHookContext = await config.preAPIHook({
+                        ...context,
+                        action: "IS_EMAIL_VERIFIED",
+                    });
+
+                    if (options === undefined || options.preAPIHook === undefined) {
+                        return postRecipeHookContext;
+                    }
+
+                    return options.preAPIHook({
+                        url: postRecipeHookContext.url,
+                        requestInit: postRecipeHookContext.requestInit,
+                    });
+                }
+            );
+
+            if (json.status === "GENERAL_ERROR") {
+                throw new Error(json.message);
+            }
+
+            if (json.status !== "OK") {
+                return {
+                    status: "CUSTOM_RESPONSE",
+                    fetchResponse,
+                };
+            }
+
+            return {
+                status: "OK",
+                isVerified: json.isVerified,
+                fetchResponse,
+            };
+        },
+
+        sendVerificationEmail: async function ({
+            config,
+            options,
+        }: {
+            config: NormalisedInputType;
+            options?: RecipeFunctionOptions;
+        }): Promise<{
+            status: "EMAIL_ALREADY_VERIFIED_ERROR" | "OK" | "CUSTOM_RESPONSE";
+            fetchResponse: Response;
+        }> {
+            const { json, fetchResponse } = await querier.post<
+                | {
+                      status: "OK" | "EMAIL_ALREADY_VERIFIED_ERROR";
+                  }
+                | APIGeneralError
+            >(SEND_VERIFY_EMAIL_PATH, {}, async (context) => {
```

Comment:
Should not be an empty body. Change the body to `{body: JSON.stringify({})}`

Reply:
Done